lab_1/main.py

```diff
@@ -86,10 +164,38 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+    if (
+        isinstance(unknown_profile, dict)
+        and isinstance(profile_to_compare, dict)
+        and isinstance(top_n, int)
+    ):
+        unk_top = get_top_n_words(unknown_profile.get("freq"), top_n)
+        check_top = get_top_n_words(profile_to_compare.get("freq"), top_n)
+        if unk_top and check_top:
+            common = [token for token in check_top if token in unk_top]
+            score = len(common) / len(check_top)
+            sorted_common = sorted(common)
+            tokens = profile_to_compare["freq"].keys()
+            max_length_word = max(tokens, key=len)
+            min_length_word = min(tokens, key=len)
+            average_token_length = sum(map(len, tokens)) / len(tokens)
```

Comment:
explain what `map` does in writing

Reply:
Here `map` applies `len` function to each element of the tokens list.